runtime/src/gpu/nvidia/gpu-nvidia.c

```diff
@@ -234,16 +241,16 @@ static void chpl_gpu_launch_kernel_help(int ln,
   CUDA_CALL(cuLaunchKernel((CUfunction)function,
                            grd_dim_x, grd_dim_y, grd_dim_z,
                            blk_dim_x, blk_dim_y, blk_dim_z,
-                           0,  // shared memory in bytes
-                           0,  // stream ID
+                           0,       // shared memory in bytes
+                           (CUstream)stream,  // stream ID
                            (void**)kernel_params,
                            NULL));  // extra options
 
   CHPL_GPU_DEBUG("cuLaunchKernel returned %s\n", name);
 
-  CUDA_CALL(cudaDeviceSynchronize());
+  chpl_task_yield();
 
-  CHPL_GPU_DEBUG("Synchronization complete %s\n", name);
+  // TODO: this is not a proper kernel_time now. What to do?
```

Comment:
Talked offline, but just for the record: I'll see if I can make this internal profile sync if it is enabled.

Otherwise, the kernel launch test measures end-to-end launch time that involves arg allocations/deallocations and also sync. So, I don't think we need to adjust the kernel launch test.